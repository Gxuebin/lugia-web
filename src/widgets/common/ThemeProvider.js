/**
 * 组件样式处理增强
 * @flow
 */
import * as React from 'react';
import PropTypes from 'prop-types';

export default (Target: React.ComponentType<any>, widgetName: string) => {
  const result = (props: Object, context: Object): React.Element<any> => {

    const getTheme = function () {
      const { config, } = context;
      const { viewClass = widgetName, } = props;
      const result = config[ viewClass ];
      return result ? result : {};
    };
    return <Target {...props} getTheme={getTheme}/>;

  };

  result.contextTypes = {
    config: PropTypes.object,
  };
  return result;
};
