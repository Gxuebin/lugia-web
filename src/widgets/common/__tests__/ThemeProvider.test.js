//@flow

import * as React from 'react';
import chai from 'chai';
import ThemeProvider from '../ThemeProvider';
import Theme from '../../theme/index';

import renderer from 'react-test-renderer';

const { expect: exp, } = chai;

describe('ThemeProvider', () => {

  it('ThemeProvider', () => {

    const TestComponent = ThemeProvider(class  extends React.Component<any> {
      render () {
        return <div>{JSON.stringify(this.props.getTheme())}{this.props.children}</div>;
      }
    }, 'ligx');

    const config = { ligx: { value: '正念', }, };
    const cmp = renderer.create(<Theme config={config}><TestComponent>hello everyone</TestComponent></Theme>);
    expect(cmp).toMatchSnapshot();
  });


});
