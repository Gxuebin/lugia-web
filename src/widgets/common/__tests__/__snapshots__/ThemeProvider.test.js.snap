// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider ThemeProvider 1`] = `
<span>
  <div>
    {"value":"正念"}
    hello everyone
  </div>
</span>
`;
